indico/modules/categories/controllers/management.py

```diff
@@ -224,3 +242,72 @@ def _process(self):
+        if request.form.get('all_selected') == 'true':
+            self.events = self.category.events
+        else:
+            event_ids = map(int, request.form.getlist('event_id'))
+            self.events = Event.query.with_parent(self.category).filter(Event.id.in_(event_ids)).all()
+
+    def _process(self):
+        is_submitted = 'submitted' in request.form
+        if not is_submitted:
+            return jsonify_template('events/management/delete_events.html', events=self.events)
+        for ev in self.events[:]:
+            delete_event(ev)
+        flash(ngettext('You have deleted one event', 'You have deleted {} events', len(self.events))
+              .format(len(self.events)), 'success')
+        return jsonify_data(flash=False, redirect=url_for('.manage_content', self.category))
+
+
+class RHSplitCategory(RHManageCategoryBase):
+    def _checkParams(self):
+        RHManageCategoryBase._checkParams(self)
+        self.category_events = {x.id: x for x in self.category.events}
+        if request.form.get('all_selected') == 'true':
+            self.event_ids = self.category_events.viewkeys()
+        else:
+            self.event_ids = set(map(int, request.form.getlist('event_id')))
+
+    def _process(self):
+        all_selected = not bool(self.category_events.viewkeys() - self.event_ids)
+        form = SplitCategoryForm(move_all=all_selected, event_id=self.event_ids)
+        if form.validate_on_submit():
+            selected_events = self.category_events.viewkeys() & self.event_ids
+            not_selected_events = self.category_events.viewkeys() - self.event_ids
+            if selected_events:
+                self._move_events(selected_events, form.first_category.data)
+            if not_selected_events:
+                self._move_events(not_selected_events, form.second_category.data)
+            return jsonify_data(flash=False, redirect=url_for('.manage_content', self.category))
+        return jsonify_form(form)
+
+    def _move_events(self, event_ids, category_title):
+        category = Category(title=category_title, parent_id=self.category.id)
+        for event_id in event_ids:
+            event = self.category_events.get(event_id)
+            category.events.append(event)
+        db.session.flush()
+
+
+class RHMoveEvents(RHManageCategoryBase):
+    def _checkParams(self):
+        RHManageCategoryBase._checkParams(self)
+        self.category_events = {x.id: x for x in self.category.events}
+        if request.json.get('all_selected'):
+            self.event_ids = self.category_events.viewkeys()
+        else:
+            self.event_ids = set(map(int, request.json.get('event_id')))
+
+    def _process(self):
+        new_category = Category.find(id=request.json.get('category_id')).first_or_404()
+        for event_id in self.event_ids:
+            if event_id in self.category_events:
+                update_event(self.category_events[event_id], {'category': new_category})
```

Comment:
I'd rather have an `event.move(category)` method for this. We probably need to trigger a signal for this operation that should be different from the "normal" even updated signal.